qldbdriver/statement_execution_test.go

```diff
@@ -448,6 +448,78 @@ func TestStatementExecution(t *testing.T) {
 		assert.True(t, ok)
 	})
 
+	t.Run("Execution metrics for stream result", func(t *testing.T) {
+		driver, err := testBase.getDriver(ledger, 10, 4)
+		require.NoError(t, err)
+		defer driver.Shutdown(context.Background())
+		defer cleanup(driver, testTableName)
+
+		// Insert docs
+		_, err = driver.Execute(context.Background(), func(txn Transaction) (interface{}, error) {
+			return txn.Execute(fmt.Sprintf("INSERT INTO %s << {'col': 1}, {'col': 2}, {'col': 3} >>", testTableName))
+		})
+		require.NoError(t, err)
+
+		selectQuery := fmt.Sprintf("SELECT * FROM %s as a, %s as b, %s as c, %s as d, %s as e, %s as f",
+			testTableName, testTableName, testTableName, testTableName, testTableName, testTableName)
+
+		_, err = driver.Execute(context.Background(), func(txn Transaction) (interface{}, error) {
+			result, err := txn.Execute(selectQuery)
+			require.NoError(t, err)
+
+			for result.Next(txn) {
+				// IOUsage test
+				ioUsage := result.GetConsumedIOs()
+				require.NotNil(t, ioUsage)
+				assert.True(t, *ioUsage.GetReadIOs() > 0)
```

Comment:
Your assertions won't run if there is no Next. How do we ensure that this got asserted

Reply:
Added `totalReadIOs` and `totalProcessingTimeMilliseconds` variables which will take the last query stats values within `Next`. We assert the variables after `Next`.